src/vendor_attribute.cpp

```diff
@@ -0,0 +1,77 @@
+#include "packet.h"
+#include "utils.h"
+#include "vendor_attribute.h"
+#include "attribute_types.h"
+
+VendorSpecific::VendorSpecific(uint8_t type, const uint8_t* data, size_t size)
+    : m_type(type),
+      m_value(size)
+{
+    m_vendorId = data[0] * (1 << 24)
```

Comment:
According to the RFC2865, high byte here must always be zero. So it might be good to check that.